bank_tiles.h

```diff
@@ -0,0 +1,52 @@
+/*
+ *  yosys -- Yosys Open SYnthesis Suite
+ *
+ *  Copyright (C) 2012  Clifford Wolf <clifford@clifford.at>
+ *  Copyright (C) 2019  The Symbiflow Authors
+ *
+ *  Permission to use, copy, modify, and/or distribute this software for any
+ *  purpose with or without fee is hereby granted, provided that the above
+ *  copyright notice and this permission notice appear in all copies.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ *  WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ *  MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ *  ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ *  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ *  ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ *  OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *
+ */
+#include "kernel/log.h"
+#include "json.hpp"
+
+#define PART_JSON "PRJXRAY_PART_JSON"
+
+USING_YOSYS_NAMESPACE
+// Coordinates of HCLK_IOI tiles associated with a specified bank
+using BankTilesMap = std::unordered_map<int, std::string>;
+
+// Find the part's JSON file with information including the IO Banks
+// and extract the bank tiles.
+BankTilesMap get_bank_tiles() {
```

Comment:
I'm not a fan of using environment variables in this way.  Can we have `write_fasm` take an argument that gets passed to `get_bank_tiles`?  Then in the top level TCL we can use the env var, but we leave it outside of the plugin.  It is generally not a good idea to proscribe a configuration strategy in the plugin.

Reply:
Yeah, we could have that. `get_bank_tiles` is also used in the xdc plugin during get_iobanks so in order to drop the env variable approach I might have to create an xdc frontend, aka `read_xdc`, which will basically behave like the `tcl` command, but will take the json argument. We could also skip the [bank check](https://github.com/SymbiFlow/yosys-symbiflow-plugins/pull/1/files/c0c9ebee7c4ae4e2ddbad434d1882c9373c7b85c#diff-bf6d652c14576b97319c59ebd0933b30R90), but I would rather keep it.

Reply:
The other option is to have a function that generates the `get_bank_tiles` data, and pass it to both `read_xdc` and `write_fasm`.  This may have utility if people want to inspect banks anyways.

Reply:
I think @litghost 's option sounds good?

Reply:
The json file can be passed using a command argument.